openwisp_monitoring/device/tests/test_api.py

```diff
@@ -291,31 +291,49 @@ def test_get_device_metrics_csv(self):
         d = self.device_model.objects.get(pk=dd.pk)
         m = self._create_object_metric(content_object=d, name='applications')
         self._create_chart(metric=m, configuration='histogram')
+        m.write(None, extra_values={'http2': 90, 'ssh': 100, 'udp': 80, 'spdy': 70})
         r = self.client.get('{0}&csv=1'.format(self._url(d.pk, d.key)))
         self.assertEqual(r.get('Content-Disposition'), 'attachment; filename=data.csv')
         self.assertEqual(r.get('Content-Type'), 'text/csv')
         rows = r.content.decode('utf8').strip().split('\n')
-        header = rows[0].strip().split(',')
-        self.assertEqual(
-            header,
-            [
-                'time',
-                'wifi_clients - WiFi clients: wlan0',
-                'wifi_clients - WiFi clients: wlan1',
-                'download - Traffic: wlan0',
-                'upload - Traffic: wlan0',
-                'download - Traffic: wlan1',
-                'upload - Traffic: wlan1',
-                'memory_usage - Memory Usage',
-                'CPU_load - CPU Load',
-                'disk_usage - Disk Usage',
-            ],
-        )
-        last_line = rows[-1].strip().split(',')
-        self.assertEqual(
-            last_line,
-            [last_line[0], '1', '2', '1.2', '0.6', '3', '1.5', '9.73', '0', '8.27'],
-        )
+        with self.subTest('Test export of all charts except histogram'):
+            header = rows[0].strip().split(',')
+            self.assertEqual(
+                header,
+                [
+                    'time',
+                    'wifi_clients - WiFi clients: wlan0',
+                    'wifi_clients - WiFi clients: wlan1',
+                    'download - Traffic: wlan0',
+                    'upload - Traffic: wlan0',
+                    'download - Traffic: wlan1',
+                    'upload - Traffic: wlan1',
+                    'memory_usage - Memory Usage',
+                    'CPU_load - CPU Load',
+                    'disk_usage - Disk Usage',
+                ],
+            )
+            last_line_before_histogram = rows[-5].strip().split(',')
+            self.assertEqual(
+                last_line_before_histogram,
+                [
+                    last_line_before_histogram[0],
+                    '1',
+                    '2',
+                    '1.2',
+                    '0.6',
+                    '3',
+                    '1.5',
+                    '9.73',
+                    '0',
+                    '8.27',
+                ],
+            )
+        with self.subTest('Test export of histogram chart'):
+            self.assertEqual(rows[-4].strip(), '')
+            self.assertEqual(rows[-3].strip(), 'Histogram')
+            self.assertEqual(rows[-2].strip().split(','), ['ssh', '100'])
+            self.assertEqual(rows[-1].strip().split(','), ['http2', '90'])
```

Comment:
There is no need for subtests in this case, just modify `last_line_before_histogram` as you already did and add the new asserts